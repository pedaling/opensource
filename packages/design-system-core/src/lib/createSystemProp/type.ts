import type { Theme } from '../types';

export type InterpolationFn = (props: Record<string, any>) => Record<string, any>;

export type SystemPropConfig = {
  property: string;
  styleProperty?: string;
  scale?: keyof Pick<Theme, 'colors' | 'space' | 'opacity'>;
  disabled?: boolean;
  transform?: (value: any) => Record<string, any> | undefined;
};

export type SystemProp = {
  (props: Record<string, any>): Record<string, any>;
  propName: string;
  disabled: boolean;
};
