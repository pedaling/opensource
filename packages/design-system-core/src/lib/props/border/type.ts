import type { ResponsiveValue } from '../../types';

export type BorderProps = {
  borderWidth?: ResponsiveValue<number>;
  borderStyle?: ResponsiveValue<'solid' | 'none'>;
  borderColor?: ResponsiveValue<string>;
  borderTopWidth?: ResponsiveValue<number>;
  borderTopStyle?: ResponsiveValue<'solid' | 'none'>;
  borderTopColor?: ResponsiveValue<string>;
  borderRightWidth?: ResponsiveValue<number>;
  borderRightStyle?: ResponsiveValue<'solid' | 'none'>;
  borderRightColor?: ResponsiveValue<string>;
  borderBottomWidth?: ResponsiveValue<number>;
  borderBottomStyle?: ResponsiveValue<'solid' | 'none'>;
  borderBottomColor?: ResponsiveValue<string>;
  borderLeftWidth?: ResponsiveValue<number>;
  borderLeftStyle?: ResponsiveValue<'solid' | 'none'>;
  borderLeftColor?: ResponsiveValue<string>;
  borderRadius?: ResponsiveValue<number | string>;
  borderTopLeftRadius?: ResponsiveValue<number | string>;
  borderTopRightRadius?: ResponsiveValue<number | string>;
  borderBottomLeftRadius?: ResponsiveValue<number | string>;
  borderBottomRightRadius?: ResponsiveValue<number | string>;
  outlineWidth?: ResponsiveValue<number>;
  outlineStyle?: ResponsiveValue<'solid' | 'none'>;
  outlineColor?: ResponsiveValue<string>;
  outlineOffset?: ResponsiveValue<number>;
};
