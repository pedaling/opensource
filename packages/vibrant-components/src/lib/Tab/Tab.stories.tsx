import type { ComponentMeta, ComponentStory } from '@storybook/react';
import { VStack } from '../VStack';
import { Tab } from './Tab';

export default {
  title: 'Tab',
  component: Tab,
  args: {
    title: 'Tab',
    active: true,
    update: true,
    id: 'tab1',
  },
} as ComponentMeta<typeof Tab>;

export const Basic: ComponentStory<typeof Tab> = props => (
  <VStack>
    <Tab {...props} />
  </VStack>
);
