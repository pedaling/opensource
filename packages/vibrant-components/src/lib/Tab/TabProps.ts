import type { ForwardedRef } from 'react';
import type { DisplaySystemProps } from '@vibrant-ui/core';
import { propVariant, withVariation } from '@vibrant-ui/core';

export type TabProps = Pick<DisplaySystemProps, 'hidden'> & {
  title: string;
  active?: boolean;
  update?: boolean;
  id: string;
  description?: string;
  onClick?: (id: string) => void;
  ref?: ForwardedRef<HTMLButtonElement>;
};

export const withTabVariation = withVariation<TabProps>('Tab')(
  propVariant({
    props: [
      {
        name: 'active',
        default: false,
      },
    ],
    variants: {
      true: {
        borderBottomColor: 'outlineNeutral',
        textColor: 'onView1',
      },
      false: {
        borderBottomColor: 'rgba(0, 0, 0, 0)',
        textColor: 'onView2',
      },
    } as const,
  })
);
