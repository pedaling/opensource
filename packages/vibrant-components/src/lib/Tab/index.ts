export { Tab } from './Tab';
export type { TabProps } from './TabProps';
