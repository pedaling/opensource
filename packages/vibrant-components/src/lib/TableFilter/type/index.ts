export type StringFilterOperator = 'contains' | 'empty' | 'equals' | 'notContains' | 'notEmpty' | 'notEquals';

export type StringFilter = {
  value: string;
  operator: StringFilterOperator[];
};

export type Filter = StringFilter & { dataKey: string };
