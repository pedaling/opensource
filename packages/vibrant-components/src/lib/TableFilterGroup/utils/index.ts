import type { DateFilterOperator, MultiSelectFilterOperator, Option, StringFilterOperator } from '../types';

export const isValueRequiredOperator = (op: string) => op === 'empty' || op === 'notEmpty';

export const isStringFilterValid = (filter: { value: string; operator: StringFilterOperator }) =>
  Boolean(filter.value) || isValueRequiredOperator(filter.operator);

export const isDateFilterValid = (filter: { value: Date[]; operator: DateFilterOperator }) => {
  if (isValueRequiredOperator(filter.operator)) return true;

  if (filter.operator === 'between') return filter.value.length >= 2;

  return filter.value.length >= 1;
};

export const isMultiSelectFilterValid = (filter: { value: Option['value'][]; operator: MultiSelectFilterOperator }) =>
  Boolean(filter.value.length > 0) || isValueRequiredOperator(filter.operator);
