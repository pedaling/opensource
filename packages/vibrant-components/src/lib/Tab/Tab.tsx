import { Box, PressableBox } from '@vibrant-ui/core';
import { Body } from '../Body';
import { NotificationBadge } from '../NotificationBadge';
import { Title } from '../Title';
import { withTabVariation } from './TabProps';

export const Tab = withTabVariation(
  ({
    innerRef,
    title,
    borderBottomColor,
    textColor,
    update,
    description,
    id,
    onClick,
    testId = 'tab',
    active,
    ...restProps
  }) => (
    <PressableBox
      role="tab"
      as="button"
      borderWidth={0}
      ref={innerRef}
      flexDirection="row"
      borderBottomWidth={2}
      borderBottomStyle="solid"
      borderBottomColor={borderBottomColor}
      position="relative"
      cursor="pointer"
      justifyContent="center"
      alignItems="center"
      onClick={() => onClick?.(id)}
      pt={12}
      pb={10}
      backgroundColor="inherit"
      ariaSelected={active}
      data-testid={testId}
      {...restProps}
    >
      <Title level={[6, 6, 5]} weight="bold" color={textColor}>
        {title}
      </Title>
      {description ? (
        <Box ml={4}>
          <Body level={[5, 5, 3]} color={textColor} weight="medium">
            {description}
          </Body>
        </Box>
      ) : null}
      {update && (
        <Box mt={2} ml={4} display="flex" alignSelf="flex-start">
          <NotificationBadge kind="dot" size="sm" />
        </Box>
      )}
    </PressableBox>
  )
);
