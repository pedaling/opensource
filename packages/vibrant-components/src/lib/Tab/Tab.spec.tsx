import userEvent from '@testing-library/user-event';
import type { ReactRenderer } from '@vibrant-ui/utils/testing';
import { createReactRenderer } from '@vibrant-ui/utils/testing';
import { Tab } from './Tab';

describe('<Tab />', () => {
  const { render } = createReactRenderer();

  let renderer: ReactRenderer;
  let element: HTMLElement;

  describe('when Tab rendered', () => {
    const mockOnClick = jest.fn(() => {});

    beforeEach(async () => {
      renderer = render(<Tab data-testid="tab" id="tab1" title="tab1" onClick={mockOnClick} />);

      element = await renderer.findByTestId('tab');
    });

    it('match snapshot', () => {
      expect(renderer.container).toMatchSnapshot();
    });

    describe('click tab', () => {
      beforeEach(async () => {
        await userEvent.click(element);
      });

      it('onClick called', () => {
        expect(mockOnClick).toBeCalledWith('tab1');
      });
    });
  });
});
