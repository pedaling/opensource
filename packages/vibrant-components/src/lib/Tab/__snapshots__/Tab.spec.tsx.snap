// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tab /> when Tab rendered match snapshot 1`] = `
.emotion-0 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border-bottom-width: 2px;
  border-bottom-style: solid;
  border-bottom-color: rgba(0, 0, 0, 0);
  position: relative;
  cursor: pointer;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 12px;
  padding-bottom: 10px;
}

.emotion-1 {
  font-family: Pretendard;
  font-size: 1rem;
  font-weight: 700;
  line-height: 1.25rem;
  color: #7C7C7C;
}

@media screen and (min-width: 1024px) {
  .emotion-1 {
    font-size: 1.125rem;
    line-height: 1.375rem;
  }
}

<div>
  <button
    class="emotion-0"
    data-testid="tab"
  >
    <span
      class="emotion-1"
    >
      tab1
    </span>
  </button>
</div>
`;
